srml/staking/reward-curve/src/log.rs

```diff
@@ -0,0 +1,67 @@
+/// Return Per-million value.
+pub fn log2(p: u32, q: u32) -> u32 {
```

Comment:
this probably should be placed in sr-primitives?

Reply:
(or now I see your assertions and comments below about how __this is not a general purpose log2__. Maybe let it be here and at some point move to sr-primitives if it became general purpose.)

Reply:
for now it is not part of runtime just compile-time.
thus I didn't completely checked that loop cannot loop forever in any situation and there is assert inside.
What I check is that if it produce something is it accurate.